src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/DefaultTagHelperContent.cs

```diff
@@ -288,14 +288,16 @@ private bool IsEmptyOrWhiteSpaceCore(object entry, EmptyOrWhiteSpaceWriter write
             var stringValue = entry as string;
             if (stringValue != null)
             {
+                // Do not encode the string because encoded value remains whitespace from user's POV.
```

Comment:
Note this branch also avoids HTML encoding. My change below just makes things consistent for strings and everything else.